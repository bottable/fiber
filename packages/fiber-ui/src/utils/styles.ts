import {
  compose,
  space,
  color,
  typography,
  height,
  SpaceProps,
  ColorProps,
  TypographyProps
} from 'styled-system'

export const styleComposition = compose(space, height, color, typography)

export type StyleProps = SpaceProps & ColorProps & TypographyProps
