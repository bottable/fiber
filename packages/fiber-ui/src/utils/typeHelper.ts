export function withProperties<A, B>(component: A, properties: B): A & B {
  Object.keys(properties).forEach((key) => {
    // @ts-ignore
    component[key] = properties[key]
  })
  return component as A & B
}
