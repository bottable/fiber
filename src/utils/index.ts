export * from './typeHelper'
export * from './mergeProps'
