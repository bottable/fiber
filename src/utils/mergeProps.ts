import React from 'react'

export type MergeElementProps<
  T extends React.ElementType,
  P extends object = {}
> = Omit<React.ComponentPropsWithRef<T>, keyof P> & P
