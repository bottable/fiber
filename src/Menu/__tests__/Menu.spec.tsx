import { Menu, Text, Divider } from '../../'

import React from 'react'
import { render } from 'test/utils'

describe('Block Menu & Menu Items', () => {
  it('renders correctly', () => {
    const block = render(
      <Menu>
        <Menu.Sub title={<span> Sub Menu 1 </span>}>
          <Menu.Sub title={<span> Sub Menu 1.1 </span>}>
            <Menu.Item>
              Item 1
            </Menu.Item>
            <Menu.Item>
              Item 2
            </Menu.Item>
          </Menu.Sub>
          <Menu.Item>
            Item 3
          </Menu.Item>
        </Menu.Sub>
        <Menu.Sub title={<span> Sub Menu 2 </span>}>
          <Menu.Item>
            Item 4
          </Menu.Item>
          <Menu.Item>
            Item 5
          </Menu.Item>
          <Menu.Item>
            Item 6
          </Menu.Item>
        </Menu.Sub>
        <Divider />
        <Menu.Item>
          Item 3
        </Menu.Item>
      </Menu>
    )

    expect(block).toMatchSnapshot()
  })
})

describe('Inline Menu & Menu Items', () => {
  it('renders correctly', () => {
    const inline = render(
      <Menu inline>
        <Menu.Item>
          <Text> Item 1 </Text>
        </Menu.Item>
        <Menu.Item>
          <Text> Item 2 </Text>
        </Menu.Item>
        <Menu.Item>
          <Text> Item 3 </Text>
        </Menu.Item>
      </Menu>
    )

    expect(inline).toMatchSnapshot()
  })
})
