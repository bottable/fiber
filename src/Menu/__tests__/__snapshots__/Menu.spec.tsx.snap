// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Block Menu & Menu Items renders correctly 1`] = `
<ul
  className="sc-AxheI cOhUdO"
>
  <li
    className="sc-Axmtr gYrhaC"
    onClick={[Function]}
  >
    <span
      className="sc-fzqyvX kTtMtC"
      size={4}
    >
       Item 1 
    </span>
  </li>
  <li
    className="sc-Axmtr gYrhaC"
    onClick={[Function]}
  >
    <span
      className="sc-fzqyvX kTtMtC"
      size={4}
    >
       Item 2 
    </span>
  </li>
  <li
    className="sc-Axmtr gYrhaC"
    onClick={[Function]}
  >
    <span
      className="sc-fzqyvX kTtMtC"
      size={4}
    >
       Item 3 
    </span>
  </li>
</ul>
`;

exports[`Inline Menu & Menu Items renders correctly 1`] = `
<ul
  className="sc-AxheI cOhUdO"
>
  <li
    className="sc-Axmtr krCCPu"
    onClick={[Function]}
  >
    <span
      className="sc-fzqyvX kTtMtC"
      size={4}
    >
       Item 1 
    </span>
  </li>
  <li
    className="sc-Axmtr krCCPu"
    onClick={[Function]}
  >
    <span
      className="sc-fzqyvX kTtMtC"
      size={4}
    >
       Item 2 
    </span>
  </li>
  <li
    className="sc-Axmtr krCCPu"
    onClick={[Function]}
  >
    <span
      className="sc-fzqyvX kTtMtC"
      size={4}
    >
       Item 3 
    </span>
  </li>
</ul>
`;
